### PR TITLE
Update Pro Scraping section

### PR DESCRIPTION
**Pull request by mstephen19**
Making changes based on @metalwarrior665's suggestions.

### CODE REVIEW

**Comment by mstephen19** [comment]
Suggestions from Lukas have been implemented. Any further pushes will be based on comments and feedback.

File: content/academy/professional_scraping/solutions/actor_building.md [hunk 1]
```diff
@@ -513,6 +500,46 @@ We've also replaced the default `title` and `description` with our own custom on
...
+
+```JavaScript
+const $ = cheerio.load(await page.content());
+```
+
+Which will allow you to use jQuery syntax outside of Puppeteer/Playwright page evaluation functions.
+
+- What is the main difference between Cheerio and jQuery?
+
+Though they share the same syntax, and therefore look quite similar, Cheerio and jQuery are designed for different purposes. jQuery is a frontend library for manipulating the DOM, making AJAX calls, etc. while Cheerio is a backend library exclusively for Node.js built for the purpose of intuitively parsing HTML on the server-side.
```
Comment (metalwarrior665) [inline content/academy/professional_scraping/solutions/actor_building.md:517]: I would add something like: `Some specific functionality of JQuery is not available with Cheerio, follow official documentation if you run into one of these cases`

File: content/academy/professional_scraping/solutions/actor_building.md [hunk 2]
```diff
@@ -513,6 +500,46 @@ We've also replaced the default `title` and `description` with our own custom on
...
+
+- In which situations would you use CheerioCrawler? What are its limitations?
+
+One should use CheerioCrawler when scraping any non-dynamic content. For scraping any content that doesn't require the loading of JavaScript in order to receive all of the data (such as with server-side rendered HTML pages and APIs), CheerioCrawler should be used. It is limited though, as it can only make static requests. This means that if a piece of data is loaded using JavaScript from an API call that the page makes, CheerioCrawler will never see that piece of data.
```
Comment (metalwarrior665) [inline content/academy/professional_scraping/solutions/actor_building.md:521]: We should mention that in the dynamic pages and API sections, we show how to overcome this limitation and enable CheerioCrawler for almost all sites

File: content/academy/professional_scraping/solutions/actor_building.md [hunk 3]
```diff
@@ -513,6 +500,46 @@ We've also replaced the default `title` and `description` with our own custom on
...
+
+> Learn more about dynamic pages in our [dynamic pages]({{@link dynamic_pages_and_spas/js_rendering_and_dynamic_pages.md}}) lesson.
+
+Additionally, if the job being done requires some sort of interaction with the page, PlaywrightCrawler/PuppeteerCrawler should be used, as CheerioCrawler runs out of the context of the browser.
+
+- Using Puppeteer, how can you extract data from a page without using jQuery/Cheerio?
+
+You can use functions such as `page.evaluate()` and `page.$$eval()` to run data collection code in the context of the browser and return it so that it can be used back in the Node.js context.
```
Comment (metalwarrior665) [inline content/academy/professional_scraping/solutions/actor_building.md:529]: Let's also add `document.querySelector` example with evaluate but mention that JQuery is preferable.

File: content/academy/professional_scraping/solutions/actor_building.md [hunk 4]
```diff
@@ -513,6 +500,46 @@ We've also replaced the default `title` and `description` with our own custom on
...
+
+- What is the default concurrency the SDK uses? Why this number?
+
+The default value for `maxConcurrency` is set to **1000**, which allows the crawler to scale up automatically to this threshold. Concurrency in the SDK is managed by the `AutoscaledPool`, which uses snapshots of the environment the actor is running in to see if it can launch more tasks than are already running (AKA **scale up**).
+
+- What is the difference between the RequestList and the RequestQueue?
+
+The main differece is that once a request list has been created, no more requests can be dynamically added to it. When you want to dynamically add (or  remove) requests, a requst queue must be used.
```
Comment (metalwarrior665) [inline content/academy/professional_scraping/solutions/actor_building.md:537]: But it is faster to add large batch of requests into and doesn't consume any platform credits.

File: content/academy/professional_scraping/solutions/integrating_webhooks.md [hunk 5]
```diff
@@ -171,11 +171,33 @@ Click **Save**, then run the Amazon **demo-actor** again.
...
 Additionally, we should be able to see that our **filter-actor** was run, and have access to its dataset:
 
-![Dataset preview]({{@asset pro_apify_scraper_development/solutions/images/dataset-preview.webp}})
+![Dataset preview]({{@asset professional_scraping/solutions/images/dataset-preview.webp}})
+
+## [](#quiz-answeers) Quiz answers 📝
+
+- How do you allocate more CPU for an actor's run?
+
+On the platform, more memory can be allocated in the actor's input configuration, and the default allocated CPU can be changed in the actor's **Settings** tab. When running locally, you can used the **APIFY_MEMORY_MBYTES** environment variable to set allocated CPU.
```
Comment (metalwarrior665) [inline content/academy/professional_scraping/solutions/integrating_webhooks.md:184]: On platform, 4 GBs equal to 1 CPU core.

File: content/academy/professional_scraping/solutions/integrating_webhooks.md [hunk 6]
```diff
@@ -171,11 +171,33 @@ Click **Save**, then run the Amazon **demo-actor** again.
...
+
+- Within itself, can you get the exact time that an actor was started?
+
+Yes. The time the actor was started can be retrieved through the `startedAt` property from the `Apify.getEnv()` function, or directly from `process.env.APIFY_STARTED_AT`
+
+- What are the types of default storages connected to an actor's run?
+
+Every actor's run is given a default key-value store, and a default dataset. The defauly key value store by default has the `INPUT` and `OUTPUT` keys.
```
Comment (metalwarrior665) [inline content/academy/professional_scraping/solutions/integrating_webhooks.md:192]: Typo is the sentence. Also mention the request queue.

File: content/academy/professional_scraping/solutions/integrating_webhooks.md [hunk 7]
```diff
@@ -171,11 +171,33 @@ Click **Save**, then run the Amazon **demo-actor** again.
...
+
+- Can you change the allocated memory of an actor while it's running?
+
+Not while it's running. You'd need to stop it and run a new one.
```
Comment (metalwarrior665) [inline content/academy/professional_scraping/solutions/integrating_webhooks.md:196]: There is also an option for soft abort and resurrect with different memory.

File: content/academy/professional_scraping/solutions/managing_source.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+## [](#quiz-answers) Quiz answers
+
+- Do you have to rebuild an actor each time the source code is changed?
+
+Yes. It needs to be built into an image, saved in a registry, and later on run in a container.
+
+- In Git, what is the difference between **pushing** changes and making a **pull request**?
+
+The pushing of code changes is usually made to a branch parallel to the one you want to eventually push it to. When creating a pull request, the code is meant to be reviewed, or at least pass all the test suites before being merged into the target branch.
```
Comment (metalwarrior665) [inline content/academy/professional_scraping/solutions/managing_source.md:17]: The first sentence is weird as it doesn't explain what pushing does. I would just tell that push changes the remote code directly but you want to usually push to a branch and then use pull request.

File: content/academy/professional_scraping/solutions/using_api_and_client.md [hunk 9]
```diff
@@ -232,6 +232,20 @@ Apify.main(async () => {
 });
 ```
 
+## [](#quiz-answers) Quiz answers 📝
+
+- What is the relationship between the Apify API and Apify client? Are there any significant differences?
+
+The Apify client mimics the Apify API, so their shouldn't be any significant differences. It's super handy as it helps with managing the API calls (parsing, error handling, retries, etc) and even adds convenience functions.
```
Comment (metalwarrior665) [inline content/academy/professional_scraping/solutions/using_api_and_client.md:239]: Would mention exponential backoff explicitly

**Review by metalwarrior665 — CHANGES_REQUESTED** [review]
Looks good, I just added several quiz improvement suggestions.

About the name, how about `Become Apify expert` or just `Apify expert course`? I just feel bad that you have to change the name on so many places, hope you have a shortcut for it :D

**Comment by mstephen19** [comment]
Made all suggested edits

File: content/academy/become_an_apify_expert.md [hunk 10]
```diff
@@ -1,13 +1,13 @@
 ---
-title: Pro Apify scraper development
+title: Become an Apify expert
```
Comment (m-murasovs) [inline content/academy/become_an_apify_expert.md:2]: Maybe this title should be `Professional scraping`? `become an Apify expert` suggests to me that the course is a continuation of the `Apify platform` course

File: content/academy/become_an_apify_expert/solutions/integrating_webhooks.md [hunk 11]
```diff
@@ -171,11 +171,33 @@ Click **Save**, then run the Amazon **demo-actor** again.
...
+![Webhook succeeded]({{@asset become_an_apify_expert/solutions/images/webhook-succeeded.webp}})
 
 Additionally, we should be able to see that our **filter-actor** was run, and have access to its dataset:
 
-![Dataset preview]({{@asset pro_apify_scraper_development/solutions/images/dataset-preview.webp}})
+![Dataset preview]({{@asset become_an_apify_expert/solutions/images/dataset-preview.webp}})
+
+## [](#quiz-answeers) Quiz answers 📝
+
+- How do you allocate more CPU for an actor's run?
```
Comment (m-murasovs) [inline content/academy/become_an_apify_expert/solutions/integrating_webhooks.md:182]: I'm not a fan of how this looks when rendered. Would you consider just making the questions bold?

Screenshot 2022-04-22 at 15 41 56

What do you think?

File: content/academy/become_an_apify_expert/solutions/managing_source.md [hunk 12]
```diff
@@ -0,0 +1,23 @@
+---
+title: Managing source code quiz answers
+description: View in-depth answers for all three of the quiz questions that were provided in the corresponding lesson about managing source code.
+menuWeight: 3
+paths:
+    - become-an-apify-expert/solutions/managing-source
+---
+
+## [](#quiz-answers) Quiz answers
```
Comment (m-murasovs) [inline content/academy/become_an_apify_expert/solutions/managing_source.md:9]: Could you make this heading consistent with the meta title? Currently, the file name, meta title, and H1 are all different.

Also, this should be H1, no?

Comment (m-murasovs) [reply content/academy/become_an_apify_expert/solutions/managing_source.md:9]: Also, if you're ok with making the questions bold in the other file, please also update here.

File: content/academy/become_an_apify_expert/solutions/using_api_and_client.md [hunk 13]
```diff
@@ -232,6 +232,22 @@ Apify.main(async () => {
 });
 ```
 
+## [](#quiz-answers) Quiz answers 📝
+
+- What is the relationship between the Apify API and Apify client? Are there any significant differences?
```
Comment (m-murasovs) [inline content/academy/become_an_apify_expert/solutions/using_api_and_client.md:237]: Same with the question formatting here

File: content/academy/become_an_apify_expert/solutions/using_storage_creating_tasks.md [hunk 14]
```diff
@@ -255,16 +242,33 @@ Don't forget to push your changes to Github using `git push origin MAIN_BRANCH_N
...
 > You can also add a custom name and description for the task in the **Settings** tab!
 
-![Creating a task]({{@asset pro_apify_scraper_development/solutions/images/creating-task.webp}})
+![Creating a task]({{@asset become_an_apify_expert/solutions/images/creating-task.webp}})
 
 After saving it, you'll be able to see the newly created task in the **Tasks** tab on the Apify Console. Go ahead and run it. Did it work?
 
+## [](#quiz-answers) Quiz answers 📝
+
+- What is the relationship between actors and tasks?
```
Comment (m-murasovs) [inline content/academy/become_an_apify_expert/solutions/using_storage_creating_tasks.md:257]: Same here for formatting

**Review by m-murasovs — CHANGES_REQUESTED** [review]

File: content/academy/become_an_apify_expert.md (hunk 10 above)
Comment (mstephen19) [reply content/academy/become_an_apify_expert.md:2]: @m-murasovs I had it as **Professional scraping**, but then @metalwarrior665 said it should be something more like **Become Apify expert**

Comment (mstephen19) [reply content/academy/become_an_apify_expert.md:2]: It is a bit of a continuation of the **Apify platform** course and the **Web scraping for beginner** course.

File: content/academy/become_an_apify_expert/solutions/integrating_webhooks.md (hunk 11 above)
Comment (mstephen19) [reply content/academy/become_an_apify_expert/solutions/integrating_webhooks.md:182]: I love this. Much better

Comment (mstephen19) [reply content/academy/become_an_apify_expert/solutions/integrating_webhooks.md:182]: I am going to make the _A:_ italic. Please don't kill me

**Comment by mstephen19** [comment]
Question formatting change. Also fixed 2 tiny linting errors in the docs file I edited in a different PR. Hope it's not too many changes :D

File: content/academy/become_an_apify_expert.md (hunk 10 above)
Comment (mstephen19) [reply content/academy/become_an_apify_expert.md:2]: cc @metalwarrior665 && @mnmkng  Thoughts?

Comment (metalwarrior665) [reply content/academy/become_an_apify_expert.md:2]: My latest suggestion was to ask inside the company, I don't want to make a singular decision about this.

Comment (mstephen19) [reply content/academy/become_an_apify_expert.md:2]: I will ask for feedback in #proj-academy

File: content/academy/become_an_apify_expert/solutions/actor_building.md [hunk 15]
```diff
@@ -502,33 +502,33 @@ Let's try it out now! Input **iphone** into the box labeled **keyword**, click *
 
 ## [](#quiz-answers) Quiz answers 📝
 
-- When using Puppeteer or Playwright, how can you still use jQuery with the SDK?
+**Q: When using Puppeteer or Playwright, how can you still use jQuery with the SDK?**
 
-There are two ways. You can either use the [injectJQuery](https://sdk.apify.com/docs/api/puppeteer#puppeteerinjectjquerypage) utility function which will enable you to use jQuery inside of `page.evalute()`, or you can use Cheerio to load the page's content like this:
+_A:_ There are two ways. You can either use the [injectJQuery](https://sdk.apify.com/docs/api/puppeteer#puppeteerinjectjquerypage) utility function which will enable you to use jQuery inside of `page.evalute()`, or you can use Cheerio to load the page's content like this:
```
Comment (m-murasovs) [inline content/academy/become_an_apify_expert/solutions/actor_building.md:507]: No need for the italics here. If anything, it can be bold?

**Review by m-murasovs — CHANGES_REQUESTED** [review]

**Review by m-murasovs — APPROVED** [review]

**Pull request merged**